### PR TITLE
added simplistic Agv support

### PR DESCRIPTION
**Pull request by liquidcronos**
## Description
In order to simulate interactions between robots and intralogistic systems a simple agv class was setup that is meant to simplify agv simulations. 
A base template class was developed for different kinematics and a differential drive kinematic childclass was implemented.

## Motivation and Context
This change is needed to simulate production cells where the workpiece is delivered by an intra logicstics system in this case an agv.

## How has this been tested?
Each class method has been unit tested against precomputed values.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: tests/test_agv_robots.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+        for target_state, correct_result in zip(target_states, correct_results):
+            target_position = target_state[0]
+            target_orientation = target_state[1]
+            self.robot.set_target_pose(target_position, target_orientation)
+
+            # calculate the position error
+            distance, angle, target_angle_error = self.robot._calculate_position_error()
+
+            # check if the position error is correct
```
Comment (MaHajo) [inline tests/test_agv_robots.py:60]: define concrete delta?

**Review by MaHajo — CHANGES_REQUESTED** [review]
Commented everything that came to mind, if you think thats bullshit feel free to ignore the comments :D
Feedback on the review itself is welcome!

File: tests/test_agv_robots.py (hunk 1 above)
Comment (liquidcronos) [reply tests/test_agv_robots.py:60]: Assert almost equal defines a standard delta up to the 7.th decimal point (scientific notation)

File: src/pybullet_industrial/agv_robots.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+        p.setJointMotorControl2(self.urdf,
+                                self.left_wheel_index,
+                                p.VELOCITY_CONTROL,
+                                targetVelocity=wheel_commands[0])
+        p.setJointMotorControl2(self.urdf,
+                                self.right_wheel_index,
+                                p.VELOCITY_CONTROL,
+                                targetVelocity=wheel_commands[1])
+
+    def standard_position_controller(self,distance,angle,target_angle_error):
```
[the remark this answers is not in the document]
Comment (liquidcronos) [reply src/pybullet_industrial/agv_robots.py:228]: added in commit 3220b467cd6d3ba1f2bfdad91fb1c1e78ef3a123

File: src/pybullet_industrial/agv_robots.py [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+        self.max_linear_velocity = diff_drive_params["max_linear_velocity"]
+        self.max_angular_velocity = diff_drive_params["max_angular_velocity"]
+
+
+        # set the control mode of the wheels to VELOCITY_CONTROL
+        p.setJointMotorControl2(self.urdf, self.left_wheel_index, p.VELOCITY_CONTROL)
+        p.setJointMotorControl2(self.urdf, self.right_wheel_index, p.VELOCITY_CONTROL)
+
+    def _calculate_wheel_comands(self, linear_velocity: float, angular_velocity: float):
+        """Calculates the wheel commands for a differential drive robot.
```
[the remark this answers is not in the document]
Comment (liquidcronos) [reply src/pybullet_industrial/agv_robots.py:181]: fixed in commit 3220b467cd6d3ba1f2bfdad91fb1c1e78ef3a123

File: src/pybullet_industrial/agv_robots.py [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+        #map angle onto the interval [-pi, pi]
+        if target_angle_error > np.pi:
+            target_angle_error -= 2*np.pi
+        elif target_angle_error < -np.pi:
+            target_angle_error += 2*np.pi
+
+        return distance_to_target, angle_to_target, target_angle_error
+
+    def standard_position_controller(self,distance,angle,target_angle_error):
```
[the remark this answers is not in the document]
Comment (liquidcronos) [reply src/pybullet_industrial/agv_robots.py:75]: fixed in commit 3220b467cd6d3ba1f2bfdad91fb1c1e78ef3a123

File: src/pybullet_industrial/agv_robots.py [hunk 5]
```diff
@@ -0,0 +1,249 @@
+"""A simple class that encapsulates mobile robots."""
```
[the remark this answers is not in the document]
Comment (liquidcronos) [reply src/pybullet_industrial/agv_robots.py]: fixed in commit 3220b467cd6d3ba1f2bfdad91fb1c1e78ef3a123

File: examples/agv_robot.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
+"""A simple class that encapsulates mobile robots."""
```
[the remark this answers is not in the document]
Comment (liquidcronos) [reply examples/agv_robot.py]: fixed in commit 3220b467cd6d3ba1f2bfdad91fb1c1e78ef3a123

File: examples/agv_robot.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+import unittest
+import os
+import pybullet_data
+
+import pybullet_industrial as pi
+import pybullet_data
+import time
+import os
+
+def trajectory_follower_controller(distance,angle,target_angle_error):
```
[the remark this answers is not in the document]
Comment (liquidcronos) [reply examples/agv_robot.py:11]: explained in commit 3220b467cd6d3ba1f2bfdad91fb1c1e78ef3a123

File: src/pybullet_industrial/agv_robots.py [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+        right_wheel_velocity = driving_contribution + turning_contribution
+
+
+        return np.array([left_wheel_velocity, right_wheel_velocity])
+
+    def set_velocity(self, velocity_vector):
+        """Sets the velocity of the robot.
+
+        Args:
+            velocity_vector (np.array): The velocity vector of the robot.
```
[the remark this answers is not in the document]
Comment (liquidcronos) [reply src/pybullet_industrial/agv_robots.py:205]: fixed in commit a43264461ae6947ac43651e880f800bc1bcc277a

**Review by MaHajo — APPROVED** [review]

**Pull request merged**